temporal-sdk/src/main/java/io/temporal/common/WorkerDeploymentVersion.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright (C) 2022 Temporal Technologies, Inc. All Rights Reserved.
+ *
+ * Copyright (C) 2012-2016 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Modifications copyright (C) 2017 Uber Technologies, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this material except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.temporal.common;
+
+import java.util.Objects;
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+/** Represents the version of a specific worker deployment. */
+@Experimental
+public class WorkerDeploymentVersion {
+  private final String deploymentName;
+  private final String buildId;
+
+  /** Build a worker deployment version from an explicit deployment name and build ID. */
+  public WorkerDeploymentVersion(@Nonnull String deploymentName, @Nonnull String buildId) {
+    this.deploymentName = deploymentName;
+    this.buildId = buildId;
+  }
+
+  /**
+   * @return The canonical string representation of this worker deployment version.
+   */
+  public String toCanonicalString() {
+    return deploymentName + "." + buildId;
+  }
```

Comment:
For this specific class, I think just having this be the `toString` instead of the generated one below would be fine

Reply:
I prefer the current approach since the canonical format is different then the standard `toString` implementation

Reply:
Meh, not sure there is such thing as a "standard" `toString` implementation, especially for a basic dot-delimited tuple like this. Same as there isn't a standard builder expectation for such a simple class. Having said that, I don't mind if we keep them separate.

Reply:
Yeah, I looked into this when I wrote it and it seems the consensus is that toString is more for debugging that canonical representations